experimental/Schemes/SpaceGerms.jl

```diff
@@ -0,0 +1,233 @@
+export SpaceGerm
+
+export representative, point
+
+export germ_at_point, ring_of_germ, ideal_of_germ, ambient_germ
+
+import AbstractAlgebra: Ring
+
+@Markdown.doc """
+    AbsSpaceGerm{BaseRingType<:Ring,RingType<:Ring}
+
+A space germ, i.e. a ringed space ``(X,O_{(X,x)})`` with local ring ``O_{(X,x)}`` of type `RingType` over a ring ``k`` of type `BaseRingType`.
+"""
+abstract type AbsSpaceGerm{BaseRingType<:Ring, RingType<:Ring} <: AbsSpec{BaseRingType, RingType} end
+
+
+####################################################################
+## two short hand definitions for internal use only
+####################################################################
+GermAtClosedPoint = Spec{<:Ring, 
```

Comment:
Maybe we want this to be a field? Or do we have germs over `ZZ`?

Reply:
You are right, this only makes sense over a field -- and currently mostly over a field of characteristic zero. But we should not enforce the latter.